app/uk/gov/hmrc/transitmovementsauditing/models/request/MetadataRequest.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovementsauditing.models.request
+
+import play.api.libs.json.Json
+import play.api.libs.json.OFormat
+import play.api.mvc.Headers
+import uk.gov.hmrc.transitmovementsauditing.config.Constants
+import uk.gov.hmrc.transitmovementsauditing.models.EORINumber
+import uk.gov.hmrc.transitmovementsauditing.models.MessageId
+import uk.gov.hmrc.transitmovementsauditing.models.MessageType
+import uk.gov.hmrc.transitmovementsauditing.models.MovementId
+import uk.gov.hmrc.transitmovementsauditing.models.MovementType
+
+object MetadataRequest {
+  implicit lazy val metadataRequestFormat: OFormat[MetadataRequest] = Json.format[MetadataRequest]
+
+  def apply(headers: Headers): MetadataRequest =
+    MetadataRequest(
+      headers.get(Constants.XAuditMetaPath).getOrElse(""),
```

Comment:
The path is mandatory -- so using `getOrElse("")` is opening up the potential for error here. There has to be a better way to do this, I'd probably argue that `buildDetails` in the controller should handle this logic (but see my other comment).